File: FE-admin/src/services/tagService.js
import tagApi from '../api/tagApi';

const tagService = {
  getAllTags: async () => {
    try {
      const response = await tagApi.getAllTags();
      return response;
    } catch (error) {
      console.error('Error fetching tags:', error);
      throw error;
    }
  },

  getTagById: async (id) => {
    try {
      const response = await tagApi.getTagById(id);
      return response;
    } catch (error) {
      console.error('Error fetching tag:', error);
      throw error;
    }
  },

  createTag: async (data) => {
    try {
      const response = await tagApi.createTag(data);
      return response;
    } catch (error) {
      console.error('Error creating tag:', error);
      throw error;
    }
  },

  updateTag: async (id, data) => {
    try {
      const response = await tagApi.updateTag(id, data);
      return response;
    } catch (error) {
      console.error('Error updating tag:', error);
      throw error;
    }
  },

  deleteTag: async (id) => {
    try {
      const response = await tagApi.deleteTag(id);
      return response;
    } catch (error) {
      console.error('Error deleting tag:', error);
      throw error;
    }
  }
};

export default tagService;

File: BE/src/main/java/com/siupo/restaurant/service/tag/TagService.java
package com.siupo.restaurant.service.tag;

import com.siupo.restaurant.dto.request.TagRequest;
import com.siupo.restaurant.dto.response.TagResponse;

import java.util.List;

public interface TagService {
    List<TagResponse> getAllTags();
    TagResponse getTagById(Long id);
    TagResponse createTag(TagRequest request);
    TagResponse updateTag(Long id, TagRequest request);
    void deleteTag(Long id);
}
